src/main/java/org/ecocean/EncounterQueryProcessor.java

```diff
@@ -338,7 +343,132 @@ public static String queryStringBuilder(HttpServletRequest request, StringBuffer
     }
     //end individualID filters-----------------------------------------------
 
+    //socialunit filters------------------------------------------------------------------
+    //community search
+     /*
+     * to filter Encounnter based on community firstly we need to get the social units
+     * associated with the communities and get the individualIDs from the social units
+     * and then use these individualIDs to filter the Encounters
+     */
+
+    if(request.getParameterValues("community")!=null){
+        String[] communities=request.getParameterValues("community");
+        int numCommunities=communities.length;
+        prettyPrint.append("Social unit is one of the following: ");
+
+        Set<String> individualsIdsSet = new HashSet<String>();
+
+        for(int i=0;i<numCommunities;i++){
+          prettyPrint.append(communities[i]+" ");
+
+          SocialUnit su =myShepherd.getSocialUnit(communities[i]);
```

Comment:
it is possible here (though, hopefully unlikely via the UI menu) that `su` is null, in the event that `communities[i]` is an invalid name, so it might be good to just skip if this is the case.